crates/hyperqueue/src/dashboard/state.rs

```diff
@@ -1,28 +1,49 @@
 use crate::dashboard::data::DashboardData;
-use crate::dashboard::ui::screen::Screen;
-use crate::dashboard::ui::screens::home::screen::HomeScreen;
+use crate::dashboard::ui::screen::{DashboardScreenState, Screen, ScreenController};
+use crate::dashboard::ui::screens::home::screen::ClusterOverviewScreen;
+use crate::dashboard::ui::screens::worker::screen::WorkerOverviewScreen;
+use std::ops::DerefMut;
 use tako::common::WrappedRcRefCell;
 
 pub struct DashboardState {
-    screens: Vec<Box<dyn Screen>>,
-    current_screen: usize, // TODO: usize newtype wrapper or enum
+    cluster_overview_screen: Box<ClusterOverviewScreen>,
```

Comment:
There's probably currently no need to box the screens if we're not keeping them inside a `Vec`.